apps/ios/SimpleX SE/ShareViewController.swift

```diff
@@ -30,12 +31,51 @@ class ShareViewController: UIHostingController<ShareView> {
                 if let error {
                     self.extensionContext!.cancelRequest(withError: error)
                 } else {
-                    self.extensionContext!.completeRequest(returningItems: [item]) { _ in
-                        let _ = sendSimpleXCmd(.apiStopChat)
+                    self.extensionContext!.completeRequest(returningItems: [item]) { expired in
+                        self.handleShutDown(expired: expired)
                     }
                 }
             }
             Task { await MainActor.run { model.item = item } }
         }
     }
+
+    override func viewDidDisappear(_ animated: Bool) {
+        // In case user dismisses the sheet, while sending a file
+        if let item = extensionContext?.inputItems.first as? NSExtensionItem {
+            if model.bottomBar.isLoading {
+                self.extensionContext!.completeRequest(returningItems: [item]) { expired in
+                    self.handleShutDown(expired: expired)
+                }
+            }
+        }
+    }
+
+    /// System will call this function twice:
+    /// - first time with `expired = false` giving time to do background processing
+    /// - second time with `expired = true` shortly before terminating the process
```

Comment:
there should be suspendChat and close database here (and it needs to be reopened on the next invocation in the same process, after call to activateChat)